sandpack-react/src/components/CodeEditor/index.tsx

```diff
@@ -102,7 +102,7 @@ export const SandpackCodeEditor = React.forwardRef<
     const { sandpack } = useSandpack();
     const { code, updateCode, readOnly: readOnlyFile } = useActiveCode();
     const { activeFile, status, editorState } = sandpack;
-    const shouldShowTabs = showTabs ?? sandpack.visibleFiles.length > 1;
+    const shouldShowTabs = showTabs ?? sandpack.visibleFiles?.length > 1;
```

Comment:
When and why `visibleFiles` becomes undefined? It should always return an array.

Reply:
> From the product point of view, I think it would be more interesting always show a file instead of an empty state. On delete, we could fall back to the next visible file, entry file, or the next file in the state. What do you think?
> 
> That said, I think your changes in this PR are still valid, but I'm also interested in how it would look for the final users.

I've reached out on twitter with a few questions, let me know if you'd prefer a github discussion etc!